rust-symcrypt/src/ecc/mod.rs

```diff
@@ -414,21 +416,16 @@ impl EcKey {
 }
 
 // Curves can be re-used across EcKey calls, creating static references to save on allocations and increase perf.
-// unwraps used here since only way this could fail is via not enough memory.
 lazy_static! {
-    static ref NIST_P256: InnerEcCurve = internal_new(CurveType::NistP256).unwrap();
-    static ref NIST_P384: InnerEcCurve = internal_new(CurveType::NistP384).unwrap();
-    static ref CURVE_25519: InnerEcCurve = internal_new(CurveType::Curve25519).unwrap();
+    static ref NIST_P256: Result<InnerEcCurve, SymCryptError> = internal_new(CurveType::NistP256);
+    static ref NIST_P384: Result<InnerEcCurve, SymCryptError> = internal_new(CurveType::NistP384);
+    static ref CURVE_25519: Result<InnerEcCurve, SymCryptError> = internal_new(CurveType::Curve25519);
 }
 
 // SymCryptInit must be called before any EcDh operations are performed.
 fn internal_new(curve: CurveType) -> Result<InnerEcCurve, SymCryptError> {
     unsafe {
         // SAFETY: FFI calls
-        // Will only init once, subsequent calls to symcrypt_init() will be no-ops.
-        // Calling here incase user did not call symcrypt_init() earlier on.
-        symcrypt_init();
```

Comment:
>symcrypt_init(); [](http://example.com/codeflow?start=8&length=16)

Why remove the init?

Reply:
Because the entry point for creating the curve has to already call `symcrypt_init()`. 

Example: 

```
 pub fn generate_key_pair(
        curve_type: CurveType,
        ec_key_usage: EcKeyUsage,
    ) -> Result<Self, SymCryptError> {
        symcrypt_init();
        let ec_curve = InnerEcCurve::new(curve_type)?;
        unsafe {
               .... 
```

I don't think there are any scenarios where you'd be able to call this without `symcrypt_init()` already being called